flutter/lib/repositories/preferences_repository.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:hive/hive.dart';
+
+abstract class PreferencesRepository {
+  Future<bool> get followSystemTheme;
+  Future<void> setFollowSystemTheme(bool value);
+}
+
+class PreferencesRepositoryImpl implements PreferencesRepository {
+  static const String kAppPreferences = "app_preferences";
+  static const String kFollowSystemKey = "key_follow_system_key";
+
+  @override
+  Future<bool> get followSystemTheme async {
+    final box = await Hive.openBox<dynamic>(kAppPreferences);
```

Comment:
why not "bool"?

Reply:
ne,  the concept is:

db named: app_preferences
key: key_follow_system_key -> true/false
or
key: "hello,world" -> int
or
key: "some key" -> "some string"

the key-value types are not fixed.